x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_review_algorithms.md

```diff
@@ -0,0 +1,239 @@
+
+#### **Scenario: Rule field is any type**
+
+**Automation**: 3 integration tests with mock rules + a set of unit tests for each algorithm
+
+```Gherkin
+Given <field_name> field is not customized by the user (current version == base version)
+And <field_name> field is updated by Elastic in this upgrade (target version != base version)
+Then for <field_name> field the diff algorithm should output the target version as the merged one without a conflict
+And <field_name> field should be returned from the `upgrade/_review` API endpoint
+And <field_name> field should be shown in the upgrade preview UI
+
+CASE: array fields should work the same agnostic of order
+
+Examples:
+| algorithm          | field_name | base_version | current_version | target_version  |
+| single line string | name       | "A"          | "A"             | "B"             |
+| number             | risk_score | 1            | 1               | 2               |
+| array of scalars   | tags       | ["A", "B"]   | ["A", "B"]      | ["A", "B", "C"] |
+```
+
+### Rule field has an update and a custom value that are the same - `ABB`
+
+#### **Scenario: Rule field is any type**
+
+**Automation**: 3 integration tests with mock rules + a set of unit tests for each algorithm
+
+```Gherkin
+Given <field_name> field is customized by the user (current version != base version)
+And <field_name> field is updated by Elastic in this upgrade (target version != base version)
+And customized <field_name> field is the same as the Elastic update in this upgrade (current version == target version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
+And <field_name> field should be returned from the `upgrade/_review` API endpoint
+And <field_name> field should be shown in the upgrade preview UI
+
+CASE: array fields should work the same agnostic of order
+
+Examples:
+| algorithm          | field_name | base_version | current_version | target_version  |
+| single line string | name       | "A"          | "B"             | "B"             |
+| number             | risk_score | 1            | 2               | 2               |
+| array of scalars   | tags       | ["A", "B"]   | ["A", "B", "C"] | ["A", "B", "C"] |
+```
+
+### Rule field has an update and a custom value that are NOT the same - `ABC`
+
+#### **Scenario: Rule field is a number or single line string**
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithms
+
+```Gherkin
+Given <field_name> field is customized by the user (current version != base version)
+And <field_name> field is updated by Elastic in this upgrade (target version != base version)
+And customized <field_name> field is different than the Elastic update in this upgrade (current version != target version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one with a conflict
```

Comment:
Do we want to mention the type of conflict (solvable / non-solvable) here or is it something that is going to be added in the later tickets?

Reply:
Yeah forgot to add those 👍